rr_iarrc/src/laplacian_line_detection/laplacian_line_detector.cpp

```diff
@@ -12,53 +12,58 @@
 #include <stdio.h>
 
 using namespace cv;
+using namespace std;
 
 cv_bridge::CvImagePtr cv_ptr;
 ros::Publisher pub, pub1, pub2, pub3;
 
 cv::Mat kernel(int, int);
 cv::Mat fillColorLines(cv::Mat, cv::Mat);
-void cutEnvironment(cv::Mat);
+void cutEnvironment(cv::Mat, int offset);
 cv::Mat cutSmall(cv::Mat, int);
 void publishMessage(ros::Publisher, Mat, std::string);
 Mat overlayBinaryGreen(Mat, Mat);
+Mat removeAngels(Mat img, int distanceFromEarth);
 
 int blockSky_height, blockWheels_height, blockBumper_height;
-int perfect_lines_min_cut, Laplacian_threshold;
+int perfect_lines_min_cut, Laplacian_threshold, adaptive_mean_threshold;
 
 
 void img_callback(const sensor_msgs::ImageConstPtr& msg) {
     cv_ptr = cv_bridge::toCvCopy(msg, "bgr8");
     Mat frame = cv_ptr->image;
 
-    //Image Height Debugger
-    //cutEnvironment(frame);
+    int orginalHeight = frame.rows;
+    int orginalWidth = frame.cols;
+    resize(frame, frame, Size(400, 400));
 
     cv::Mat frame_gray, frame_blur, detected_edges;
-    GaussianBlur(frame, frame_blur, Size(5,5), 0, 0, BORDER_DEFAULT);
+    GaussianBlur(frame, frame_blur, Size(5,5), 0);
     cv::cvtColor(frame_blur, frame_gray, cv::COLOR_BGR2GRAY);
 
     Mat thres;
     adaptiveThreshold(frame_gray, thres, 255, ADAPTIVE_THRESH_GAUSSIAN_C, THRESH_BINARY, 5, -1);
-    cutEnvironment(thres);
-    Mat cut = cutSmall(thres, perfect_lines_min_cut);   //Make sure only Lines
+    cutEnvironment(thres, 0);
+    erode(thres, thres, kernel(3,1));
 
     Mat lapl;
     Laplacian(frame_gray, lapl, CV_16S, 3, 1, 0, BORDER_DEFAULT);
-    lapl = min(lapl, 0);
+    threshold(lapl, lapl, -Laplacian_threshold, 255, 1);
     convertScaleAbs(lapl, lapl);
-    threshold(lapl, lapl, Laplacian_threshold, 255, 0);
 
+    Mat cut = cutSmall(thres, perfect_lines_min_cut);
     Mat fill = fillColorLines(lapl, cut);
     fill = cutSmall(fill, perfect_lines_min_cut);
 
-    Mat green_lines = overlayBinaryGreen(frame, fill);
+    resize(fill, fill, Size(orginalWidth, orginalHeight));
 
-//	publish images
     publishMessage(pub, fill, "mono8");
-    publishMessage(pub1, cut, "mono8");
+    publishMessage(pub1, thres, "mono8");
     publishMessage(pub2, lapl, "mono8");
-    publishMessage(pub3, green_lines, "bgr8");
+
+//    Uncomment for Green Overlay
+//    Mat green_lines = overlayBinaryGreen(frame, fill);
+//    publishMessage(pub3, green_lines, "bgr8");
```

Comment:
This might be a good place for checking pub3.getNumSubscribers() and publishing if > 0